core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -237,8 +345,76 @@ class SpendReportingService(
        | AND $timePartitionColumn BETWEEN @startDate AND @endDate
        | AND project.id in UNNEST(@projects)
        | GROUP BY currency ${queryKeys.map(_.bigQueryGroupByClause()).mkString}
-       |""".stripMargin
-      .replace("REPLACE_TIME_PARTITION_COLUMN", timePartitionColumn)
+       |""".stripMargin.replace("REPLACE_TIME_PARTITION_COLUMN", timePartitionColumn)
+  }
+
+  private def getTimePartitionColumn(tableName: String): String = {
+    val isBroadTable = tableName == spendReportingServiceConfig.defaultTableName
+    // The Broad table uses a view with a different column name.
+    if (isBroadTable) spendReportingServiceConfig.defaultTimePartitionColumn else "_PARTITIONTIME"
+  }
+
+  def getAllUserWorkspaceQuery(
+    billingProjects: Map[BillingProjectSpendExport, Seq[(GoogleProjectId, WorkspaceName)]],
+    pageSize: Int,
+    offset: Int
+  ): String = {
+    val baseQuery = s"""
+                       |  SELECT
+                       |    project.id AS project_id,
+                       |    project.name AS project_name,
+                       |    currency,
+                       |    SUM(IFNULL((SELECT SUM(c.amount) FROM UNNEST(credits) c), 0)) as credits,
```

Comment:
do `credits` also have a description like "Cloud Storage" or "Compute Engine" we could use to categorize them?

Reply:
I don't know and I don't have the permissions to figure it out

Reply:
I do see a `type` column defined for the `credits` record. If this `CASE … WHEN` doesn't work out, we could look at the `type` column. I think we'll really just need to test this empirically and see what results it gives.